tensorflow_model_optimization/python/core/common/keras/compression/algorithm.py

```diff
@@ -110,6 +110,23 @@ def training(self, *training_weights: tf.Tensor) -> tf.Tensor:
        tf.Tensor to set the compressible weight to.
     """
 
+  def get_compressible_weights(
+      self, original_layer: tf.keras.layers.Layer) -> List[str]:
+    """Define compressible weights for each layer.
+
+    Args:
+       original_layer: tf.keras.layers.Layer representing any layers from
+         the original model.
+
+    Returns:
+       List of strings representing list of compressible weights for the given
+       layer.
+    """
+    if isinstance(original_layer, tf.keras.layers.Conv2D) or \
```

Comment:
This is probably okay to start for usability testing but beyond that

1) Providing defaults like this is not desirable. For instance, if the Keras library added a new built-in layer with weights, this would automatically make all TFMOT libraries built on this API indicate that they support this new layer when the algorithm developer may not have considered it yet. The default is set to [], when may be the algorithm developer wants to compress the weights. If the algorithm developer decides to compress the weights and modifies the default, that then silently causes model developer's models to become more aggressively optimized (e.g. resulting in potential accuracy reductions) and is in a sense, a backward-incompatible change. This idea should be elaborating in some of the older pruning design docs. 

2)  We should remember to consider the tradeoff of including it as a part of the algorithm API versus separately passing in an object into an API (e.g. `create_layer_for_training`). In the current quantization API, the two are separate.

Reply:
For 2. it'll probably become more obvious once we consider aspects like handling custom Keras layers, etc.

Reply:
Applied option 1 and added a comment for 2. I agree that we have to consider how we can supports custom keras layer.

Reply:
For 1. returning the [] as the default set of weights to optimize leads to the same issue. 

Nit: "representing any layers from the original model" -> "representing a layer from the original model"

Nit: "List of strings representing list of compressible weights for the given layer." 
-> Would be good to reduce the thinking by connecting it to "attributes" and perhaps providing an example (e.g. 'kernel').

Reply:
Okay. I got it. I prefer to give autonomy to algorithm developer.
For now, I suggest that algorithm developer should add param flags on their algorithm to control compressible weights if their algorithm needed.
But it's good to make good practice for that configuration, or provide some common utils.